### PR TITLE
DROOLS-6559 DMN put() put all() context()

### PR DESCRIPTION
**Pull request by tarilabs**
**JIRA**: https://issues.redhat.com/browse/DROOLS-6559

**referenced Pull Requests**: none

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tarilabs** [comment]
@Rikkola or @karreiro , can either of you kindly review specifically commit eaf911f please?

In short I don't believe this reflection access to be needed on GWT, as I would expect all calling FEEL functions to not even reach this class and navigate pure Maps, but please let me know

/cc @danielezonca

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/model/SupportRequest.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
+package org.kie.dmn.feel.model;
```
Comment (danielezonca) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/model/SupportRequest.java:1]: License header

**Comment by tarilabs** [comment]
```
16:18:39  Body did not finish within grace period; terminating with extreme prejudice
[Pipeline] }
[Pipeline] // script
Error when executing always post condition:
org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
	at org.jenkinsci.plugins.workflow.steps.TimeoutStepExecution.cancel(TimeoutStepExecution.java:165)
```

🤦 

Jenkins retest this

**Review by hellowdan — APPROVED** [review]

**Review by karreiro — APPROVED** [review]
Thank you, @tarilabs.

The PR is great and it introduces no issues from FEEL IntelliSense perspective. However, I have a question for @Rikkola (considering the https://github.com/kiegroup/drools/pull/3566 effort).

@Rikkola, I believe we're planning to extract all GWT-uncompatible implementations to utility classes, so we may maintain a single implementation of FEEL. With that in mind, I believe the plan is to remove all `UnsupportedOperationException` occurrences from `ImmutableFPAWrappingPOJO.java` (in the `kie-dmn-feel-gwt`) to ensure that single implementation. Is my understanding correct?

**Comment by tarilabs** [comment]
Thanks for the review @karreiro in https://github.com/kiegroup/drools/pull/3794#pullrequestreview-738387043

**Review by danielezonca — APPROVED** [review]

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/model/SupportRequest.java (hunk 1 above)
Comment (tarilabs) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/model/SupportRequest.java:1]: fair, will do in another PR

Comment (tarilabs) [reply kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/model/SupportRequest.java:1]: done with 

* https://github.com/kiegroup/drools/pull/3800

**Comment by Rikkola** [comment]
@karreiro @tarilabs Yes, I will adjust the clean up/gwt feel functions PR to support these changes once this is in.

**Comment by tarilabs** [comment]
It is already "in" :) merged with 2c90e48 on main 🚀

**Pull request merged**